packages/babel-core/src/parse.js

```diff
@@ -39,9 +39,15 @@ export const parse: Parse = (function parse(code, opts, callback) {
     opts = undefined;
   }
 
-  // For backward-compat with Babel 7's early betas, we allow sync parsing when
-  // no callback is given. Will be dropped in some future Babel major version.
-  if (callback === undefined) return parseRunner.sync(code, opts);
+  if (callback === undefined) {
+    const message =
+      "Starting from Babel 8.0.0, the 'parse' function expects a callback. If you need to call it synchronously, please use 'parseSync'.";
+    if (process.env.BABEL_8_BREAKING) {
+      throw new Error(message);
+    } else {
+      console.warn(message);
```

Comment:
We should still call `parseRunner.sync` here.

Reply:
🤔Shouldn't `parseRunner.errback(code, opts, undefined)` fallback to `parserRunner.sync`?